core/src/main/java/org/apache/iceberg/AllManifestsTable.java

```diff
@@ -158,12 +168,14 @@ static class ManifestListReadTask implements DataTask {
     private DataFile lazyDataFile = null;
 
     ManifestListReadTask(
+        Schema dataTableSchema,
```

Comment:
needs the table schema to be able to parse partition spec.

the other arg `schema` is for the `MANIFEST_FILE_SCHEMA`.